### PR TITLE
cromwell metadata params [risk: low]

### PR DESCRIPTION
**Pull request by davidangb**
DataBiosphere/firecloud-app#213
DataBiosphere/firecloud-app#214

when making requests to the Cromwell workflow metadata API (via orch and rawls), specify only those `includeKey` values the UI needs. This hint allows Cromwell to do only the work that's truly necessary, and keeps the response sizes smaller by eliminating cruft.

additionally, don't load the data needed to power the timing diagram when first viewing a workflow's details, since the timing diagram is collapsed by default. Once the user clicks to expand the timing diagram, make a second ajax call to get the data it needs. This keeps the initial page much smaller and only gets timing  data when absolutely necessary.

-----

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/main/broadfcui/endpoints.cljs [hunk 1]
```diff
@@ -140,7 +141,7 @@
              (set (keys query-parameters)))
           (str "Malformed query parameters: " query-parameters))
   {:path (str "/workspaces/" (id-path workspace-id) "/entityQuery/" type "?"
-              (clojure.string/join "&" (keep (fn [[k v]]
+              (string/join "&" (keep (fn [[k v]]
                                                (some->> v str not-empty (str k "=")))
                                              query-parameters)))
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/endpoints.cljs]: reindent

File: src/cljs/main/broadfcui/page/workspace/monitor/workflow_details.cljs [hunk 2]
```diff
@@ -82,34 +89,65 @@
...
-                               (.timingDiagram js/window chart-container data workflow-name 100)
-                               (let [height (getTimingDiagramHeight chart-container)]
-                                 (.timingDiagram js/window chart-container data workflow-name (+ 75 height)))))))
-               :path "https://www.gstatic.com/charts/loader.js"}])
-           [:div {:ref "chart-container" :style {:paddingTop "0.5rem"}}]])]))})
+        (create-field label
+          (links/create-internal {:onClick #(swap! state update :expanded? not)}
+            (if expanded? "Hide" "Show")))
+        ;; use expanded? to show/hide the WorkflowTimingDiagram component via the DOM
+        [:div {:myattr "whereami" :style {:display (if expanded? "block" "none")}}
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/page/workspace/monitor/workflow_details.cljs]: debugging artifact?

**Review by MattPutnam — APPROVED** [review]
👍

File: src/cljs/main/broadfcui/page/workspace/monitor/workflow_details.cljs (hunk 2 above)
Comment (davidangb) [reply src/cljs/main/broadfcui/page/workspace/monitor/workflow_details.cljs]: hahaha yes thank you for the catch.

**Pull request merged**